tests/Fixer/ArrayNotation/AlphabeticalArrayKeySortFixerTest.php

```diff
@@ -0,0 +1,270 @@
+    }
+
+    /**
+     * @return iterable<string, array{0: string, 1: string, 2?: array<string, string>}>
+     */
+    public static function provideFixCases(): iterable
+    {
+        yield 'array with special case keys on bottom' => [
+            '<?php [
+                    "a.*" => "int",
+                    "b" => "array",
+                    "c.*" => "array",
+                    "d" => "int",
+                    "d.*" => "string",
+                    "b" . "a" => "bla",
+                    "a" . "a" => "bla",
+                ];',
+            '<?php [
+                    "c.*" => "array",
+                    "b" => "array",
+                    "d.*" => "string",
+                    "b" . "a" => "bla",
+                    "a" . "a" => "bla",
+                    "d" => "int",
+                    "a.*" => "int",
+                ];',
+            [
+                'special_keys_placement' => 'bottom',
+            ],
+        ];
+
+        yield 'array with special case keys on top' => [
+            '<?php [
+                    "b" . "a" => "bla",
+                    "a" . "a" => "bla",
+                    "a.*" => "int",
+                    "b" => "array",
+                    "c.*" => "array",
+                    "d" => "int",
+                    "d.*" => "string",
+                ];',
+            '<?php [
+                    "c.*" => "array",
+                    "b" => "array",
+                    "d.*" => "string",
+                    "b" . "a" => "bla",
+                    "a" . "a" => "bla",
+                    "d" => "int",
+                    "a.*" => "int",
+                ];',
+            [
+                'special_keys_placement' => 'top',
+            ],
+        ];
+
+        yield 'old-style array with string keys' => [
+            '<?php array("a" => "a", "b" => "b", "c" => "c");',
+            '<?php array("c" => "c", "a" => "a", "b" => "b");',
+        ];
+
+        yield 'old-style array with integer keys' => [
+            '<?php array(1 => "one", 2 => "two", 3 => "three");',
+            '<?php array(3 => "three", 1 => "one", 2 => "two");',
+        ];
+
+        yield 'old-style array with mixed keys' => [
+            '<?php array("a" => "a", "b" => "b", 1 => "one", 2 => "two");',
+            '<?php array("b" => "b", 1 => "one", 2 => "two", "a" => "a");',
+        ];
+
+        yield 'array with string keys' => [
+            '<?php ["a" => "a", "b" => "b", "c" => "c"];',
+            '<?php ["c" => "c", "a" => "a", "b" => "b"];',
+        ];
+
+        yield 'array with mixed keys, starting with integer' => [
+            '<?php ["a" => "a", "b" => "b", "c" => "c", 0 => 0, 1 => 1, 2 => 2];',
+            '<?php [2 => 2, "c" => "c", 0 => 0, "a" => "a", "b" => "b", 1 => 1];',
+        ];
+
+        yield 'array with integer keys' => [
+            '<?php [1 => "one", 2 => "two", 3 => "three"];',
+            '<?php [3 => "three", 1 => "one", 2 => "two"];',
```

Comment:
please those cases, I think they are not handled well:

1,11,2 vs 1,2,11
```php
[1 => "a", 10 => "j", 2 => "b", 6 => "f", 11 => "k" ];
```

I did not found a case like this already
```php
[1 => "a", 10 => "j", "b", 6 => "f", 11 => "k" ]
```